`carthage archive <NAME> --project-directory <DIR>` ignores <DIR> value

* `carthage version`: 0.16.2
* `xcodebuild -version`: Xcode 7.3.1 (Build version 7D1014)
* Are you using `--no-build`? No
* Are you using `--no-use-binaries`? No
* Are you using `--use-submodules`? No

**Cartfile**
```
EMPTY
```

**Carthage Output**
```
See below for some output in context
```

In relation to https://github.com/Carthage/Carthage/issues/1347 I decided to try moving `PusherSwift.xcodeproj` into a directory so that it existed at `PusherSwift/PusherSwift.xcodeproj`.

I then ran `carthage build --no-skip-current --project-directory PusherSwift` and everything works as expected.

Running `carthage archive PusherSwift` then can't find the frameworks, as expected I suppose.

So I ran it with the project directory specified as in the build command, so like this:

```
carthage archive PusherSwift --project-directory PusherSwift
```

But this still said that there were no frameworks.

I cloned Carthage and started adding logging into `Archive.Swift` and found that it doesn't seem to currently take the `options.directoryPath` into account. I could make the archive command work with my code by changing this line: https://github.com/Carthage/Carthage/blob/4cbbe5dbeb916cccbeca3fd52a6b76f743cae5e2/Source/carthage/Archive.swift#L81

to this:

```swift
return NSFileManager.defaultManager().currentDirectoryPath.stringByAppendingString("/\(options.directoryPath)/").stringByAppendingString((platform.relativePath as NSString).stringByAppendingPathComponent(framework))
```

Am I going crazy or is that indeed something like what is needed?

Obviously if someone specifies an absolute path then the code I changed it to wouldn't work, so maybe you'd need something a bit more like this:

```swift
return SignalProducer(values: frameworks).map { framework in
  let currentPath = NSFileManager.defaultManager().currentDirectoryPath
  let relativePath: String

  if (NSFileManager.defaultManager().fileExistsAtPath(options.directoryPath)) {
    return options.directoryPath.stringByAppendingString((platform.relativePath as NSString).stringByAppendingPathComponent(framework))
  } else {
    let directoryPathRelativeToCurrentPathString = currentPath.stringByAppendingString("/\(options.directoryPath)/")
    let resolvedPath = NSURL(fileURLWithPath: directoryPathRelativeToCurrentPathString).URLByStandardizingPath?.path
    if resolvedPath == nil {
      return resolvedPath
    } else {
      return resolvedPath!.stringByAppendingString((platform.relativePath as NSString).stringByAppendingPathComponent(framework))
    }
  }
}
```

Again, this is a bit pseudocode-y, but you get the idea.

The problem with the above (when I tested a simpler version) is that I get this error when then running `make install`:

```
CompileSwift normal x86_64
    cd /Users/Hami/Pusher/Carthage
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift -frontend -c /Users/Hami/Pusher/Carthage/Source/carthage/Formatting.swift /Users/Hami/Pusher/Carthage/Source/carthage/Fetch.swift /Users/Hami/Pusher/Carthage/Source/carthage/CopyFrameworks.swift /Users/Hami/Pusher/Carthage/Source/carthage/Bootstrap.swift /Users/Hami/Pusher/Carthage/Source/carthage/Version.swift /Users/Hami/Pusher/Carthage/Source/carthage/Update.swift /Users/Hami/Pusher/Carthage/Source/carthage/Extensions.swift /Users/Hami/Pusher/Carthage/Source/carthage/Environment.swift /Users/Hami/Pusher/Carthage/Source/carthage/Build.swift /Users/Hami/Pusher/Carthage/Source/carthage/Outdated.swift /Users/Hami/Pusher/Carthage/Source/carthage/main.swift /Users/Hami/Pusher/Carthage/Source/carthage/Archive.swift /Users/Hami/Pusher/Carthage/Source/carthage/Checkout.swift -target x86_64-apple-macosx10.10 -enable-objc-interop -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -I /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/BuildProductsPath/Release -F /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/BuildProductsPath/Release -g -module-cache-path /Users/Hami/Library/Developer/Xcode/DerivedData/ModuleCache -serialize-debugging-options -Xcc -I/Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/swift-overrides.hmap -Xcc -iquote -Xcc /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/carthage-generated-files.hmap -Xcc -I/Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/carthage-own-target-headers.hmap -Xcc -I/Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/carthage-all-non-framework-target-headers.hmap -Xcc -ivfsoverlay -Xcc /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/all-product-headers.yaml -Xcc -iquote -Xcc /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/carthage-project-headers.hmap -Xcc -I/Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/BuildProductsPath/Release/include -Xcc -I/Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/DerivedSources/x86_64 -Xcc -I/Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/DerivedSources -Xcc -DNDEBUG=1 -Xcc -working-directory/Users/Hami/Pusher/Carthage -emit-module-doc-path /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/Objects-normal/x86_64/carthage.swiftdoc -O -module-name carthage -emit-module-path /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/Objects-normal/x86_64/carthage.swiftmodule -emit-objc-header-path /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/Objects-normal/x86_64/carthage-Swift.h -serialize-diagnostics-path /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/Objects-normal/x86_64/Formatting.dia -emit-dependencies-path /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/Objects-normal/x86_64/Formatting.d -num-threads 4 -o /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/Objects-normal/x86_64/Formatting.o -o /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/Objects-normal/x86_64/Fetch.o -o /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/Objects-normal/x86_64/CopyFrameworks.o -o /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/Objects-normal/x86_64/Bootstrap.o -o /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/Objects-normal/x86_64/Version.o -o /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/Objects-normal/x86_64/Update.o -o /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/Objects-normal/x86_64/Extensions.o -o /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/Objects-normal/x86_64/Environment.o -o /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/Objects-normal/x86_64/Build.o -o /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/Objects-normal/x86_64/Outdated.o -o /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/Objects-normal/x86_64/main.o -o /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/Objects-normal/x86_64/Archive.o -o /Users/Hami/Library/Developer/Xcode/DerivedData/Carthage-ezcnplddawcrewateqwnsbitvvwv/Build/Intermediates/ArchiveIntermediates/carthage/IntermediateBuildFilesPath/Carthage.build/Release/carthage.build/Objects-normal/x86_64/Checkout.o
/Users/Hami/Pusher/Carthage/Source/carthage/Archive.swift:85:62: error: value of type '[Platform]' has no member 'relativePath'
                                                        return options.directoryPath.stringByAppendingString((platform.relativePath as NSString).stringByAppendingPathComponent(framework))
                                                                                                              ^~~~~~~~ ~~~~~~~~~~~~
```

I think this is probably just because I don't have any experience with ReactiveCocoa and have no idea why platform seems to be both a `Platform` and a `[Platform]` at the same time ❓

--- a/Source/carthage/Archive.swift
+++ b/Source/carthage/Archive.swift
@@ -78,7 +78,7 @@ public struct ArchiveCommand: CommandType {
 			return SignalProducer(values: Platform.supportedPlatforms)
 				.flatMap(.Merge) { platform -> SignalProducer<String, CarthageError> in
 					return SignalProducer(values: frameworks).map { framework in
-						return (platform.relativePath as NSString).stringByAppendingPathComponent(framework)
+						return (options.directoryPath as NSString).stringByAppendingPathComponent((platform.relativePath as NSString).stringByAppendingPathComponent(framework))
 					}
 				}
 				.filter { relativePath in NSFileManager.defaultManager().fileExistsAtPath(relativePath) }

